MapboxDirections/Attributes/MBSpeedLimit.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+
+/**
+ A localized limit for measuring speed limits.
+ */
+@objc(MBSpeedLimit)
+public class SpeedLimit: NSObject, NSSecureCoding {
+    
+    /**
+     A unitless measure of speed which is dependent on the `MaximumSpeedLimit.speedUnits`.
+     
+     If a speed is not known, it will default to `Int.NSNotFound`.
+     */
+    @objc public var speed: Int = NSNotFound
+    
+    
+    /**
+     Units for `MaximumSpeedLimit.speed`.
+     */
+    @objc public var speedUnits: SpeedUnit = .none
+    
+    
+    /**
+     `Bool` whether the maximum speed for a segment is known.
+     */
+    @objc public var speedIsUnknown: Bool = true
```

Comment:
A separate property would be a suboptimal way to represent an unknown value. `value == -1` is already sufficient to indicate that the speed limit is unknown. So we can formalize that fact: define a constant `MBSpeedUnknown` that has a `value` of –1. This is how `CLLocationCoordinate2D` and `kCLLocationCoordinate2DInvalid` work.

Down the line, if we need more symbolic values in the future, such as `variable`, then we can define an enumeration for those symbolic values and add a field of that type to the `Speed` struct. That would be useful for distinguishing between legal and advisory speed limits, for example.